File: backend/schemas/informes.py
from datetime import datetime
from pydantic import BaseModel, ConfigDict
from decimal import Decimal

class ProductoCatalogoVistaOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    id_producto: int
    nombre: str
    descripcion: str | None
    precio_venta: Decimal
    precio_compra: Decimal
    cant_disponible: int
    id_categoria: int
    categoria_nombre: str
    categoria_descripcion: str | None

class VentaPorCategoriaOut(BaseModel):
    id_categoria: int
    categoria_nombre: str
    num_ventas: int
    costo_total: Decimal
    factura_promedio_categoria: Decimal | None

class ProductoNuncaVendidoOut(BaseModel):
    id_producto: int
    nombre: str
    cant_disponible: int

class VentaPorMesSubqueryOut(BaseModel):
    anio_mes: str
    cantidad_ventas: int
    total_mes: Decimal

class UltimaLineaVentaOut(BaseModel):
    id_venta: int
    fecha: datetime
    cliente: str
    producto: str
    cantidad_venta: int
    precio_venta: Decimal


class TopProductoVendidoOut(BaseModel):
    id_producto: int
    nombre: str
    total_unidades_vendidas: int
    costo_aproximado: Decimal


class CompraPorMesOut(BaseModel):
    anio_mes: str
    num_compras: int
    total_mes: Decimal


class VentaPorEmpleadoOut(BaseModel):
    nit_empleado: str | None
    empleado_nombre: str | None
    num_ventas: int
    costo_total: Decimal


class VentaRangoOut(BaseModel):
    id_venta: int
    fecha: datetime
    total: Decimal
    cliente: str | None
    empleado: str | None

File: frontend/src/types/informes.ts
export type ProductoCatalogoVistaRow = {
  id_producto: number;
  nombre: string;
  descripcion: string | null;
  precio_venta: string | number;
  precio_compra: string | number;
  cant_disponible: number;
  id_categoria: number;
  categoria_nombre: string;
  categoria_descripcion: string | null;
};

export type VentaPorCategoriaRow = {
  id_categoria: number;
  categoria_nombre: string;
  num_ventas: number;
  costo_total: string | number;
  factura_promedio_categoria: string | number | null;
};

export type ProductoNuncaVendidoRow = {
  id_producto: number;
  nombre: string;
  cant_disponible: number;
};

export type VentaPorMesRow = {
  anio_mes: string;
  cantidad_ventas: number;
  total_mes: string | number;
};

export type UltimaLineaVentaRow = {
  id_venta: number;
  fecha: string;
  cliente: string;
  producto: string;
  cantidad_venta: number;
  precio_venta: string | number;
};

export type TopProductoVendidoRow = {
  id_producto: number;
  nombre: string;
  total_unidades_vendidas: number;
  costo_aproximado: string | number;
};

export type CompraPorMesRow = {
  anio_mes: string;
  num_compras: number;
  total_mes: string | number;
};

export type VentaPorEmpleadoRow = {
  nit_empleado: string | null;
  empleado_nombre: string | null;
  num_ventas: number;
  costo_total: string | number;
};
